### ISSUE TITLE
ans_loss and hop_loss became 'nan'

### ISSUE THREAD

**Issue by michaelmoju** [THUDM/CogQA#10]
Hello, I followed all the instructions to train the System1 model. Since I have only 1 RTX2080 GPU, I adjusted the batch size to 4 instead of 12. That is `python train.py --batch_size 4`. After training about 60% of the training process, the hop_loss and ans_loss became `nan`. I used another server which has multiple GPUs and trained system1 with the default settings `batch_size=12` and it works successfully. I wonder whether it is caused by the loss function `LogSoftmax` which may be `-inf`? Do you have any solution for this problem? Thank you.

**Comment by Sleepychord** [comment]
I am not about the reasons. Sometimes Pytorch will come across this kind of issues. Please give me more details to reproduce if you are still interested in it.
